packages/web/src/components/send-tokens-modal/SendTokensModal.tsx

```diff
@@ -0,0 +1,181 @@
+import { useState } from 'react'
+
+import {
+  useArtistCoin,
+  transformArtistCoinToTokenInfo,
+  useSendTokens
+} from '@audius/common/api'
+import { AUDIO, FixedDecimal } from '@audius/fixed-decimal'
+
+import ResponsiveModal from 'components/modal/ResponsiveModal'
+
+import SendTokensConfirmation from './SendTokensConfirmation'
+import SendTokensFailure from './SendTokensFailure'
+import SendTokensInput from './SendTokensInput'
+import SendTokensProgress from './SendTokensProgress'
+import SendTokensSuccess from './SendTokensSuccess'
+import { SendTokensModalProps, SendTokensState } from './types'
+
+const SendTokensModal = ({
+  mint,
+  onClose,
+  walletAddress,
+  isOpen
+}: SendTokensModalProps) => {
+  const [state, setState] = useState<SendTokensState>({
+    step: 'input',
+    amount: BigInt(0),
+    destinationAddress: ''
+  })
+  const [error, setError] = useState<string>('')
+
+  // Get token data and balance using the same hooks as ReceiveTokensModal
+  const { data: coin } = useArtistCoin({ mint })
+  const tokenInfo = coin ? transformArtistCoinToTokenInfo(coin) : undefined
+
+  // Use the new tan-query hook for sending tokens
+  const sendTokensMutation = useSendTokens({ mint })
+
+  const handleInputContinue = (amount: bigint, destinationAddress: string) => {
+    setState({
+      step: 'confirm',
+      amount,
+      destinationAddress
+    })
+  }
+
+  const handleConfirm = async () => {
+    setState((prev) => ({ ...prev, step: 'progress' }))
+    setError('') // Clear any previous errors
+
+    try {
+      // Use the new hook to send tokens
+      await sendTokensMutation.mutateAsync({
+        recipientWallet: state.destinationAddress as any, // Type assertion for now
+        amount: AUDIO(state.amount).value // Convert bigint to AudioWei
+      })
+
+      // If successful, move to success step
+      setState((prev) => ({ ...prev, step: 'success' }))
```

Comment:
is this something we could do in the hook itself? or is that too web-specific?

Reply:
ohh good q, ill consider it. my guess is that we want to do a different pattern for mobile since itll be screen based.